marshmallow_recipe/serialization.py

```diff
@@ -151,3 +151,15 @@ def dump_many_v2(data: list[Any], *, naming_case: NamingCase | None = None) -> l
     dump_many = dump_many_v2
 
 EmptySchema = m.Schema
+
+
+def _extract_type(data: Any, cls: type | None) -> type:
+    if hasattr(data, "__orig_class__"):
+        return getattr(data, "__orig_class__")
+    data_type = type(data)
+    if not cls or cls == data_type:
+        return data_type
+    origin = get_origin(cls)
+    if origin != data_type:
+        raise ValueError(f"{cls=} is not subscripted version of {data_type}")
```

Comment:
Am I right that this error happens when you don't pass `cls` to `dump` when it's needed?

Reply:
no. please read section "Extracting generic instance type" from PR description and check `test_generic_extract_type_on_dump` to clear understanding which cases are valid and which will raise an error snd which error exactly 

https://github.com/anna-money/marshmallow-recipe/pull/163/files/68aff75faecf53833c455309671a03150117d432#diff-7f5a7271c4eef44b2f5fda6334ecae0952550de169520dde4a919d86949361c2R659